apps/examples/board_specific_demo/ameba_spi_lcd_demo/example_spi_lcd.c

```diff
@@ -0,0 +1,152 @@
+#include <tinyara/config.h>
+#include <tinyara/spi/spi.h>
+#include <tinyara/lcd/lcd.h>
+#include <tinyara/lcd/lcd_dev.h>
+#include <sys/types.h>
+#include <stdint.h>
+#include <stdbool.h>
+#include <unistd.h>
+#include <string.h>
+#include <fcntl.h>
+#include <poll.h>
+#include <errno.h>
+#include <debug.h>
+#include <mqueue.h>
+#include <stdio.h>
+#include <stdlib.h>
+#  define ST7789_XRES       240
+#  define ST7789_YRES       320
+
+#define LCD_DEV_PATH "/dev/lcd%d"
+static pid_t g_task = 0;
+
+void test_put_area()
+{
+	int fd = 0, p = 0, count = 0;
+	char port[20] = {'\0'};
+	sprintf(port, LCD_DEV_PATH, p);
+	fd = open(port, O_RDWR | O_SYNC, 0666);
+	if (fd < 0) {
+		printf("ERROR: Failed to open lcd port : %s error:%d\n", port, fd);
+		return -1;
+	}
+	struct lcddev_area_s area;
+	area.row_start = 10;
+	area.row_end = 19;
+	area.col_start = 10;
+	area.col_end = 19;
+	area.stride = 10*2;
+	area.planeno = 0;
+	int i=0;
+	uint8_t spi_data[401];
+	for(i=0;i<=100;i+=2)
+	{
+  		spi_data[i] = 0x07E0 & 0X00FF;
+  		spi_data[i+1] = (0x07E0 & 0xFF00)>>8; 
+	}
+	for(i=100;i<=200;i+=2)
+	{
+		spi_data[i] = 0x001F & 0X00FF;
+  		spi_data[i+1] = (0x001F & 0xFF00)>>8; 
+	}
+	for(i=200;i<=400;i+=2)
+	{
+		spi_data[i] = 0xF800 & 0X00FF;
+  		spi_data[i+1] = (0xF800 & 0xFF00)>>8; 
+	}
```

Comment:
There are many hard coded values above. The meaning of these values is not clear. 
Please add preprocessor definitions or comments to indicate the meaning of each value.